From 9edfba34203531aedc070f1caa73870758e6d726 Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Mon, 3 Mar 2014 09:26:21 +0000
Subject: [PATCH 139/351] mfd: florida: Mark DSP clocking registers as
 volatile

The DSPs will often control there own clock speeds whilst running as
such we should mark the registers controlling this as volatile.

Change-Id: I7ac81c88f93a4f26bd5eb62b5b33a001f2b86f5b
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 drivers/mfd/florida-tables.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/mfd/florida-tables.c b/drivers/mfd/florida-tables.c
index b7c4ffe..9195e51 100644
--- a/drivers/mfd/florida-tables.c
+++ b/drivers/mfd/florida-tables.c
@@ -2697,6 +2697,7 @@ static bool florida_volatile_register(struct device *dev, unsigned int reg)
 	case ARIZONA_DSP1_SCRATCH_1:
 	case ARIZONA_DSP1_SCRATCH_2:
 	case ARIZONA_DSP1_SCRATCH_3:
+	case ARIZONA_DSP1_CLOCKING_1:
 	case ARIZONA_DSP2_STATUS_1:
 	case ARIZONA_DSP2_STATUS_2:
 	case ARIZONA_DSP2_STATUS_3:
@@ -2725,6 +2726,7 @@ static bool florida_volatile_register(struct device *dev, unsigned int reg)
 	case ARIZONA_DSP2_SCRATCH_1:
 	case ARIZONA_DSP2_SCRATCH_2:
 	case ARIZONA_DSP2_SCRATCH_3:
+	case ARIZONA_DSP2_CLOCKING_1:
 	case ARIZONA_DSP3_STATUS_1:
 	case ARIZONA_DSP3_STATUS_2:
 	case ARIZONA_DSP3_STATUS_3:
@@ -2753,6 +2755,7 @@ static bool florida_volatile_register(struct device *dev, unsigned int reg)
 	case ARIZONA_DSP3_SCRATCH_1:
 	case ARIZONA_DSP3_SCRATCH_2:
 	case ARIZONA_DSP3_SCRATCH_3:
+	case ARIZONA_DSP3_CLOCKING_1:
 	case ARIZONA_DSP4_STATUS_1:
 	case ARIZONA_DSP4_STATUS_2:
 	case ARIZONA_DSP4_STATUS_3:
@@ -2781,6 +2784,7 @@ static bool florida_volatile_register(struct device *dev, unsigned int reg)
 	case ARIZONA_DSP4_SCRATCH_1:
 	case ARIZONA_DSP4_SCRATCH_2:
 	case ARIZONA_DSP4_SCRATCH_3:
+	case ARIZONA_DSP4_CLOCKING_1:
 		return true;
 	default:
 		return florida_is_adsp_memory(dev, reg);
-- 
1.7.9.5

